x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/convert/AbstractConvertFunction.java

```diff
@@ -78,7 +79,15 @@ protected TypeResolution resolveType() {
     }
 
     public Set<DataType> supportedTypes() {
-        return factories().keySet();
+        Set<DataType> supportedTypes = new HashSet<>();
```

Comment:
it's either this check here or adding a capabilities check for each function/operator on the evaluators list.
if that's the case we should probably do the same for `date_nanos` as well.
but I preferred this check than modifying the evaluators list.